lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+                         .removeClass('hidden')
+                         .focus();
+                     this.toggleSessionSelectionPanel();
+                 },
+
+                 unenrollSuccess: function() {
+                     /*
+                     Update external elements on the course card to represent the unenrolled state.
+
+                     1) Hide the change session button and the date field.
+                     2) Hide the 'View Course' button.
+                     3) Remove the messages associated with the enrolled state.
+                     4) Remove the link from the course card image and title.
+                     */
+
+                     // Update the model with the new session Id;
+                     this.entitlementModel.set({currentSessionId: this.currentSessionSelection});
+
+                     // Reset the card contents to the unenrolled state
+                     this.$triggerOpenBtn.addClass('hidden');
+                     this.$enterCourseBtn.addClass('hidden');
+                     this.$courseCardMessages.remove();
+                     this.$('.enroll-btn-initial').focus();
+                     HtmlUtils.setHtml(
+                        this.$dateDisplayField,
+                         HtmlUtils.joinHtml(
+                             HtmlUtils.HTML('<span class="icon fa fa-warning" aria-hidden="true"></span>'),
+                             HtmlUtils.HTML(gettext('You must select a session below to access course.'))
+                         )
+                     );
+
+                     // Remove links to previously enrolled sessions
+                     this.$courseImageLink.replaceWith( // xss-lint: disable=javascript-jquery-insertion
+                         HtmlUtils.joinHtml(
+                             HtmlUtils.HTML('<div class="'),
+                             this.$courseImageLink.attr('class'),
+                             HtmlUtils.HTML('" tabindex="-1">'),
+                             HtmlUtils.HTML(this.$courseImageLink.html()),
+                             HtmlUtils.HTML('</div>')
+                         ).text
+                     );
+                     this.$courseTitleLink.replaceWith( // xss-lint: disable=javascript-jquery-insertion
+                         HtmlUtils.joinHtml(
+                             HtmlUtils.HTML('<span>'),
+                             this.$courseTitleLink.text(),
+                             HtmlUtils.HTML('</span>')
+                         ).text
+                     );
+                 },
+
+                 enrollError: function() {
+                     var errorMsgEl = gettext('There was an error, please reload the page and try again.');
+                     this.$dateDisplayField
+                         .find('.fa.fa-spin')
+                         .removeClass('fa-spin fa-spinner')
+                         .addClass('fa-close');
+                     this.$dateDisplayField.append(errorMsgEl);
```

Comment:
Will errorMsgEl be escaped if it's appended like this?

Reply:
My understanding is that gettext deals with the escaping

Reply:
fixed